src/storage/segcache/src/segcache.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! Core datastructure
+
+use crate::*;
+
+const RESERVE_RETRIES: usize = 3;
```

Comment:
What's your empirical observation of this value? Does it happen rarely, no so rarely? Is 3 considered "plenty safe" or "you might want to tune this if you have a small-ish heap"?

Reply:
A single retry is used if eviction needs to occur - the remaining two retries would only occur if eviction fails somehow. In a non concurrent implementation, additional retries should be very rare - but could theoretically occur under high eviction pressure where we have a stale list of "least valuable segments" (~1s TTL on that list).

I'll add a stat here so we can evaluate, my gut says this is "plenty safe" and helps set an upper bound on insert latency - but having some data to back the default would be nice